packages/assets-controllers/src/TokensController.ts

```diff
@@ -844,6 +905,62 @@ export class TokensController extends BaseController<
   clearIgnoredTokens() {
     this.update({ ignoredTokens: [], allIgnoredTokens: {} });
   }
+
+  _requestApproval(
+    suggestedAssetMeta: SuggestedAssetMeta & {
+      interactingAddress: string;
+    },
+  ) {
+    this.messagingSystem
+      .call(
+        'ApprovalController:addRequest',
+        {
+          id: suggestedAssetMeta.id,
+          origin: ORIGIN_METAMASK,
+          type: APPROVAL_TYPE,
+          requestData: {
```

Comment:
For mobile, we will need requestData to be populated with the following values, as it currently relies on the `suggestedAssetMeta` object sent by the `pendingSuggestedAsset` event.

Unfortunately, the type `SuggestedAssetMeta` is not compatible with `requestData` as it has several optional and an unknown field within the `Token` type, so we are sending precisely the data it needs.